p2p/src/net/default_backend/peer.rs

```diff
@@ -121,28 +125,49 @@ where
             peer_event_tx,
             backend_event_rx,
             node_protocol_version,
+            time_getter,
         }
     }
 
     fn validate_peer_time(
         p2p_config: &P2pConfig,
-        local_time: Duration,
-        remote_time: Duration,
+        local_time: std::ops::RangeInclusive<Time>,
```

Comment:
IMO `local_time` that is a range looks rather cryptic. I'd suggest to make it two separate parameters with more descriptive names.